utils/resultstable.go

```diff
@@ -1048,3 +1060,50 @@ func extractDependencyNameFromComponent(key string, techIdentifier string) (depe
 	dependencyName = split[0]
 	return
 }
+
+func getApplicabilityStatusFromRule(rule *sarif.ReportingDescriptor) ApplicabilityStatus {
+	if rule.Properties["applicability"] != nil {
+		status, ok := rule.Properties["applicability"].(string)
+		if !ok {
+			log.Debug(fmt.Sprintf("failed to get applicability status from rule properties for rule_id %s", rule.ID))
+		}
+		switch status {
+		case "not_covered":
+			return NotCovered
+		case "undetermined":
+			return ApplicabilityUndetermined
+		case "not_applicable":
+			return NotApplicable
+		case "applicable":
+			return Applicable
+		}
+	}
+	return ""
```

Comment:
```suggestion
	if status, ok := rule.Properties["applicability"].(string); ok {
		switch status {
        case "not_covered":
            return NotCovered
        case "undetermined":
            return ApplicabilityUndetermined
        case "not_applicable":
            return NotApplicable
        case "applicable":
            return Applicable
        }
	} else {
        log.Debug(fmt.Sprintf("Failed to get applicability status from rule properties for rule_id %s", rule.ID))
    }
	return NotScanned
```
Maybe the check for nil is redundent, if checking for string is also handling the nil case.
Comments/Logs should stat with Capital.

Reply:
@attiasas It looks like `rule.Properties["applicability"].(string)` checks both whether the key exists in the map and whether its value is of the expected type - but it considered good practice to check for the existence of a key before attempting to access it.